### PR TITLE
Added Name of current fragment in AppBar

### LINKED ISSUE #630: AppBar doesn't show name of current activity/fragment
how user will know which option he has selected
suppose user has selected Add Password option and user will redirected to the activity/fragment
but on the AppBar it should show name of fragment

### Steps to reproduce
if user has selected Add Password option and user will redirected to the activity/fragment

### Expected behaviour
Name of current fragment should be in AppBar, because user can know what is the purpose of current screen

### Would you like to work on the issue?
Yes

### PR DESCRIPTION
**Pull request by bharatmk257**
# Description

Added title on every activity/fragment.

change

Fixes #630

## Type of change
Just put an x in the [] which are valid.
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
- [ ] `./gradlew assembleDebug assembleRelease`
- [ ] `./gradlew checkstyle`

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by vyankatesh24** [comment]
@bharatmk257 
There are conflicts in PR. First Pull from the master branch and then make your changes.

File: app/src/main/java/swati4star/createpdf/activity/MainActivity.java [hunk 1]
```diff
@@ -369,87 +371,109 @@ public boolean onNavigationItemSelected(@NonNull MenuItem item) {
         switch (item.getItemId()) {
             case R.id.nav_home:
                 fragment = new HomeFragment();
+                setTitle(R.string.app_name);
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/activity/MainActivity.java]: Instead of having this in every switch-case, can we have a hashmap and just use that.
Also, you havent changed the title for `nav_add_watermark`

**Comment by Swati4star** [comment]
Checks are failing.

File: app/src/main/java/swati4star/createpdf/activity/MainActivity.java (hunk 1 above)
Comment (bharatmk257) [reply app/src/main/java/swati4star/createpdf/activity/MainActivity.java]: Sorry i didn't get you

**Comment by bharatmk257** [comment]
i have removed conflicts

File: app/src/main/java/swati4star/createpdf/activity/MainActivity.java (hunk 1 above)
Comment (sidhuparas) [reply app/src/main/java/swati4star/createpdf/activity/MainActivity.java]: Check out the `fillMap()` method of `HomeFragment` class. That should be enough to make Swati's comment clear to you.

**Comment by vyankatesh24** [comment]
@bharatmk257 
The `title` is not changing if I opens a `Fragment` using `Buttons` on `Home Screen`
It is changing only if the `Navigation Drawer` is used.

**Comment by Swati4star** [comment]
@sidhuparas @vyankatesh24 Review this one

**Comment by sidhuparas** [comment]
@bharatmk257 Open app. Click "View Files". Press back button. Click "History". Title doesn't change. Click back button. Now open any fragment and the title doesn't change.

File: app/src/main/java/swati4star/createpdf/activity/MainActivity.java [hunk 2]
```diff
@@ -79,6 +79,7 @@
     private Fragment mCurrentFragment;
 
 
+
```
Comment (sidhuparas) [inline app/src/main/java/swati4star/createpdf/activity/MainActivity.java]: Remove this excess empty line.

File: app/src/main/java/swati4star/createpdf/activity/MainActivity.java [hunk 3]
```diff
@@ -24,6 +24,7 @@
 import android.widget.Toast;
 
 import java.util.ArrayList;
+import java.util.HashMap;
```
Comment (sidhuparas) [inline app/src/main/java/swati4star/createpdf/activity/MainActivity.java]: This is unnecessary import. Remove it.

File: app/src/main/java/swati4star/createpdf/activity/MainActivity.java [hunk 4]
```diff
@@ -293,6 +295,7 @@ public void onBackPressed() {
                 getSupportFragmentManager().beginTransaction().replace(R.id.content, fragment).commit();
                 setTitle(R.string.app_name);
                 setNavigationViewSelection(R.id.nav_home);
+
```
Comment (sidhuparas) [inline app/src/main/java/swati4star/createpdf/activity/MainActivity.java]: Remove this empty line as well.

**Comment by bharatmk257** [comment]
@sidhuparas 
there is problem with ViewFilesFragment.java file

i have checked in line number-351 :- actionBar.setTitle(appName);

when i checked appName is final value 

and no need to setting title if i remove this line it will solve problem 

i am removing line number 350 and 351

**Comment by sidhuparas** [comment]
@bharatmk257 Nice bug fix. It's working fine now. One thing needs to be changed now. `Share` and `Help` shouldn't change the title as they don't open any section. Also, pull the latest code that includes `Whats New`. Title for it should also not change for obvious reasons.

File: app/src/main/java/swati4star/createpdf/activity/MainActivity.java [hunk 5]
```diff
@@ -77,7 +77,6 @@
     private boolean mDoubleBackToExitPressedOnce = false;
     private Fragment mCurrentFragment;
 
-
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/activity/MainActivity.java]: revert back this change

File: app/src/main/java/swati4star/createpdf/fragment/ViewFilesFragment.java [hunk 6]
```diff
@@ -347,8 +347,6 @@ public void onStop() {
         super.onStop();
         AppCompatActivity activity = ((AppCompatActivity)
                 Objects.requireNonNull(mActivity));
-        ActionBar actionBar = activity.getSupportActionBar();
```
[the remark this answers is not in the document]
Comment (bharatmk257) [reply app/src/main/java/swati4star/createpdf/fragment/ViewFilesFragment.java]: okay i did added this line

File: app/src/main/java/swati4star/createpdf/fragment/HomeFragment.java [hunk 7]
```diff
@@ -141,80 +141,99 @@ public void onClick(View v) {
         switch (v.getId()) {
             case R.id.images_to_pdf:
                 fragment = new ImageToPdfFragment();
+                getActivity().setTitle(R.string.images_to_pdf);
```
[the remark this answers is not in the document]
Comment (bharatmk257) [reply app/src/main/java/swati4star/createpdf/fragment/HomeFragment.java]: okay

File: app/src/main/java/swati4star/createpdf/activity/MainActivity.java (hunk 5 above)
Comment (bharatmk257) [reply app/src/main/java/swati4star/createpdf/activity/MainActivity.java]: in line 80 ??
it was just clear space

Comment (bharatmk257) [reply app/src/main/java/swati4star/createpdf/activity/MainActivity.java]: i did set app name in home fragment because 
when user will go to another fragment it will change title
after he can get into home fragment with tow way 
1 is back button = it will change title to app name automatically 
2 is via navigation drawer = it's not changing title so i have used setTitle

File: app/src/main/java/swati4star/createpdf/activity/MainActivity.java (hunk 1 above)
Comment (bharatmk257) [reply app/src/main/java/swati4star/createpdf/activity/MainActivity.java]: how can i use hashmap for setting title 
i didn't understand can you please share some reference for this

**Comment by bharatmk257** [comment]
i have committed this solution do check it
i think it will solve both problems

File: app/src/main/java/swati4star/createpdf/fragment/ViewFilesFragment.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (Swati4star) [reply app/src/main/java/swati4star/createpdf/fragment/ViewFilesFragment.java]: you have commented the code

File: app/src/main/java/swati4star/createpdf/activity/MainActivity.java (hunk 1 above)
Comment (Swati4star) [reply app/src/main/java/swati4star/createpdf/activity/MainActivity.java]: @sidhuparas Help @bharatmk257

Comment (sidhuparas) [reply app/src/main/java/swati4star/createpdf/activity/MainActivity.java]: @bharatmk257 As I wrote in one of the comments above `fillMap()` method of `HomeFragment` class is perfect example. Check that code and ask me any specific question.

**Comment by sidhuparas** [comment]
@bharatmk257 What problem are you facing now? I have already provided you the `HashMap` hint.

**Comment by bharatmk257** [comment]
I know abkut hashmap and i have used it but i am still nit getting how can i set title via hashmap

Can you please share some docs or any reference for that

**Comment by sidhuparas** [comment]
@bharatmk257 I have already shared. Have you taken a look at: https://github.com/Swati4star/Images-to-PDF/pull/633#discussion_r269524943

File: app/src/main/java/swati4star/createpdf/activity/MainActivity.java [hunk 8]
```diff
@@ -512,20 +514,35 @@ public void setNavigationViewSelection(int id) {
...
-        return permission;
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            if ((ContextCompat.checkSelfPermission(this, Manifest.permission.WRITE_EXTERNAL_STORAGE)
+                    != PackageManager.PERMISSION_GRANTED) &&
+                    (ContextCompat.checkSelfPermission(this, Manifest.permission.READ_EXTERNAL_STORAGE)
+                            != PackageManager.PERMISSION_GRANTED)) {
+                return false;
+            }
+        }
+        return true;
```
Comment (sidhuparas) [inline app/src/main/java/swati4star/createpdf/activity/MainActivity.java]: Why have you changed permissions code?

File: app/src/main/java/swati4star/createpdf/activity/MainActivity.java [hunk 9]
```diff
@@ -512,20 +514,35 @@ public void setNavigationViewSelection(int id) {
         mNavigationView.setCheckedItem(id);
     }
 
-    private void getRuntimePermissions() {
-        PermissionsUtils.checkRuntimePermissions(this,
-                PERMISSION_REQUEST_CODE,
-                Manifest.permission.WRITE_EXTERNAL_STORAGE,
-                Manifest.permission.READ_EXTERNAL_STORAGE,
-                Manifest.permission.CAMERA);
```
Comment (sidhuparas) [inline app/src/main/java/swati4star/createpdf/activity/MainActivity.java]: Revert this as well.

File: app/src/main/java/swati4star/createpdf/activity/MainActivity.java [hunk 10]
```diff
@@ -369,6 +373,31 @@ private void checkDoubleBackPress() {
...
+        mFragmentPositionMap.append(R.id.nav_extract_images, R.string.extract_images);
+        mFragmentPositionMap.append(R.id.nav_pdf_to_images, R.string.pdf_to_images);
+        mFragmentPositionMap.append(R.id.nav_remove_pages, R.string.remove_pages);
+        mFragmentPositionMap.append(R.id.nav_rearrange_pages, R.string.reorder_pages);
+        mFragmentPositionMap.append(R.id.nav_compress_pdf, R.string.compress_pdf);
+        mFragmentPositionMap.append(R.id.nav_add_images, R.string.add_images);
+        mFragmentPositionMap.append(R.id.nav_remove_duplicate_pages, R.string.remove_duplicate_pages);
+        mFragmentPositionMap.append(R.id.nav_invert_pdf, R.string.invert_pdf);
+        mFragmentPositionMap.append(R.id.nav_add_watermark, R.string.add_watermark);
+        mFragmentPositionMap.append(R.id.nav_zip_to_pdf, R.string.zip_to_pdf);
```
Comment (sidhuparas) [inline app/src/main/java/swati4star/createpdf/activity/MainActivity.java]: Nice work on HashMap implementation 👍

File: app/src/main/java/swati4star/createpdf/activity/MainActivity.java (hunk 8 above)
Comment (bharatmk257) [reply app/src/main/java/swati4star/createpdf/activity/MainActivity.java]: i am still thinking when i changed this :thinking: 
when i have merged with swati master channel it was automatically added to my code 

lemme change it

File: app/src/main/java/swati4star/createpdf/activity/MainActivity.java [hunk 11]
```diff
@@ -368,6 +373,31 @@ private void checkDoubleBackPress() {
         this.mDoubleBackToExitPressedOnce = true;
         Toast.makeText(this, R.string.confirm_exit_message, Toast.LENGTH_SHORT).show();
     }
+    private void fillMap() {
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/activity/MainActivity.java]: Add javadoc style comments to newly added functions

File: app/src/main/java/swati4star/createpdf/activity/MainActivity.java [hunk 12]
```diff
@@ -493,20 +523,35 @@ public void setNavigationViewSelection(int id) {
         mNavigationView.setCheckedItem(id);
     }
 
-    private void getRuntimePermissions() {
-        PermissionsUtils.checkRuntimePermissions(this,
-                PERMISSION_REQUEST_CODE,
-                Manifest.permission.WRITE_EXTERNAL_STORAGE,
-                Manifest.permission.READ_EXTERNAL_STORAGE,
-                Manifest.permission.CAMERA);
+    private boolean getRuntimePermissions() {
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/activity/MainActivity.java]: there are a number of unnecessary changes. Revert those.

After creating a PR, you should see the "files changed" section. If there is something you didnt change, look into it

File: app/src/main/java/swati4star/createpdf/fragment/HomeFragment.java [hunk 13]
```diff
@@ -145,83 +145,103 @@ public void onClick(View v) {
...
                 fragment.setArguments(bundle);
+                mActivity.setTitle(R.string.add_images);
                 break;
             case R.id.remove_duplicates_pages_pdf:
                 fragment = new RemoveDuplicatePagesFragment();
+                mActivity.setTitle(R.string.remove_duplicate_pages);
                 break;
             case R.id.invert_pdf:
                 fragment = new InvertPdfFragment();
+                mActivity.setTitle(R.string.invert_pdf);
```
Comment (Swati4star) [inline app/src/main/java/swati4star/createpdf/fragment/HomeFragment.java]: if we have hashmap now, why do we need these lines then?

**Comment by bharatmk257** [comment]
i am done with it but just one problem i am facing i can't use if statement if i use build is failing 

**Checkstyle rule violations were found**

**Comment by vyankatesh24** [comment]
> i am done with it but just one problem i am facing i can't use if statement if i use build is failing
> 
> **Checkstyle rule violations were found**

./gradlew checkstyle

Run this command locally in android studio terminal
you will get to see what the error is.

**Comment by bharatmk257** [comment]
Checkstyle rule violations were found. See the report at: file:///home/bharatmk257/Documents/AndroidStudioProjects/Images-to-PDF/app/build/reports/checkstyle/checkstyle.html

same error while executing on Command line

File: app/src/main/java/swati4star/createpdf/activity/FavouritesActivity.java [hunk 14]
```diff
@@ -39,6 +39,7 @@
     @Override
     protected void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        setTitle("Add to Favourite");
```
Comment (vyankatesh24) [inline app/src/main/java/swati4star/createpdf/activity/FavouritesActivity.java]: Put this string in `strings.xml`

File: app/src/main/java/swati4star/createpdf/activity/MainActivity.java (hunk 2 above)
Comment (vyankatesh24) [reply app/src/main/java/swati4star/createpdf/activity/MainActivity.java]: @bharatmk257 Have a look!

File: app/src/main/java/swati4star/createpdf/activity/MainActivity.java [hunk 15]
```diff
@@ -161,6 +164,7 @@ public boolean onOptionsItemSelected(MenuItem item) {
         if (item.getItemId() == R.id.menu_favourites_item) {
             Fragment fragment = new FavouritesFragment();
             FragmentManager fragmentManager = getSupportFragmentManager();
+            setTitle("Favourites");
```
Comment (vyankatesh24) [inline app/src/main/java/swati4star/createpdf/activity/MainActivity.java]: Put this string in `strings.xml`

File: app/src/main/java/swati4star/createpdf/activity/MainActivity.java (hunk 4 above)
Comment (vyankatesh24) [reply app/src/main/java/swati4star/createpdf/activity/MainActivity.java]: have a look!

**Comment by vyankatesh24** [comment]
@bharatmk257 
Congratulations on your first contribution to Images-to-PDF ! :+1:

**Comment by bharatmk257** [comment]
Bhai Maza aa Gaya :heart_eyes:
